### PR TITLE
Allow missing authdata checksum with GSSAPI

### PR DESCRIPTION
**Pull request by simo5**
Allow missing checksum for compatibility with bad SMB clients

### CODE REVIEW

File: src/lib/gssapi/krb5/accept_sec_context.c [hunk 1]
```diff
@@ -670,13 +670,15 @@ kg_accept_krb5(minor_status, context_handle,
...
 
     if (authdat->checksum == NULL) {
-        /* missing checksum counts as "inappropriate type" */
-        code = KRB5KRB_AP_ERR_INAPP_CKSUM;
-        major_status = GSS_S_FAILURE;
-        goto fail;
-    }
+        /* Some SMB client implementations use handcrafted GSSAPI code
+         * that does not provide a checksum. MS-KILE documents that the
+         * Microsoft implementation considers a missign checksum acceptable
```
Comment (abbra) [inline src/lib/gssapi/krb5/accept_sec_context.c]: typo: missign -> missing

Comment (simo5) [inline src/lib/gssapi/krb5/accept_sec_context.c]: On Tue, 2015-08-04 at 11:21 -0700, Alexander Bokovoy wrote:
> > @@ -670,13 +670,15 @@ kg_accept_krb5(minor_status, context_handle,
> >  #endif
> >  
> >      if (authdat->checksum == NULL) {
> > -        /* missing checksum counts as "inappropriate type" */
> > -        code = KRB5KRB_AP_ERR_INAPP_CKSUM;
> > -        major_status = GSS_S_FAILURE;
> > -        goto fail;
> > -    }
> > +        /* Some SMB client implementations use handcrafted GSSAPI code
> > +         * that does not provide a checksum. MS-KILE documents that the
> > +         * Microsoft implementation considers a missign checksum acceptable
> 
> typo: missign -> missing
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/krb5/krb5/pull/306/files#r36223030

Sigh

thanks

-- 
Simo Sorce * Red Hat, Inc * New York

**Comment by simo5** [comment]
Fixed some typos

**Comment by greghudson** [comment]
Pushed to master with minor changes in 0e60d5ce041607cfc7659a8d3198d0f3f8958245

**Pull request closed without merging**